functests/ptp/prometheus.go

```diff
@@ -0,0 +1,126 @@
+package ptp
+
+import (
+	"encoding/json"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	client "github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const openshiftPtpNamespace = "openshift-ptp"
+const openshiftPtpMetricPrefix = "openshift_ptp_"
+const openshiftMonitoringNamespace = "openshift-monitoring"
+
+type queryOutput struct {
+	Data data
+}
+
+type data struct {
+	Result []result
+}
+
+type result struct {
+	Metric metric
+}
+
+type metric struct {
+	Pod string
+}
+
+var _ = Describe("prometheus", func() {
+	Context("Metrics reported by PTP pods", func() {
+		It("Should all be reported by prometheus", func() {
+			ptpPods, err := client.Client.Pods(openshiftPtpNamespace).List(metav1.ListOptions{
+				LabelSelector: "app=linuxptp-daemon",
+			})
+			Expect(err).ToNot(HaveOccurred())
+			ptpMonitoredEntriesByPod, uniqueMetricKeys := collectPtpMetrics(ptpPods.Items)
+			podsPerPrometheusMetricKey := collectPrometheusMetrics(uniqueMetricKeys)
+			checkIfPrometheusMetricsContainAllPtpEntries(ptpMonitoredEntriesByPod, podsPerPrometheusMetricKey)
+		})
+	})
+})
+
+func checkIfPrometheusMetricsContainAllPtpEntries(ptpMetricsByPod map[string][]string, prometheusMetrics map[string][]string) {
```

Comment:
Can this be something more generic like `containSameMetrics`? 
Also, I'd move this to the bottom of the file since this is less important then the other two util functions.

Reply:
Done